datafusion/functions-nested/src/extract.rs

```diff
@@ -200,6 +199,7 @@ fn array_element_inner(args: &[ArrayRef]) -> Result<ArrayRef> {
     let [array, indexes] = take_function_args("array_element", args)?;
 
     match &array.data_type() {
+        Null => Ok(Arc::new(NullArray::new(array.len()))),
```

Comment:
can we return the null with List type?

Reply:
I will write a test that shows the issue. But basically something like `array_union(make_array(), ['x', 'y', 'z'])` will be an issue because we can't union a list of ints and a list of strings.

Reply:
Or in this case similar nested expression with array element

Reply:
I added this query as a test:
`select coalesce(array_element([], 1), array_element(NULL, 1), 'ok');`